prime-router/src/main/kotlin/fhirengine/translation/HL7toFhirTranslator.kt

```diff
@@ -24,11 +25,49 @@ class HL7toFhirTranslator(
     private val messageEngine: HL7MessageEngine = FhirTranscoder.getMessageEngine(),
 ) : Logging {
     companion object {
+        /**
+         * List of all available mapping directory locations
+         */
+        internal val configPaths: List<String> = listOf(
+            "./metadata/HL7/catchall",
```

Comment:
All permissible mapping file locations are now listed here and will be initialized at construction sequentially. This change is due to the message template retrieval process relying on a singleton instance of `ResourceReader` that could potentially be redirected to another mapping location if multiple mapping file locations were retrieved concurrently.